src/lib/sidenav/sidenav.ts

```diff
@@ -161,7 +162,7 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
     let activeEl = this._doc && this._doc.activeElement;
     if (activeEl && this._elementRef.nativeElement.contains(activeEl)) {
       if (this._elementFocusedBeforeSidenavWasOpened instanceof HTMLElement) {
-        this._elementFocusedBeforeSidenavWasOpened.focus();
+        this._focusOriginMonitor.focusVia(this._elementFocusedBeforeSidenavWasOpened, 'program');
```

Comment:
I think the closing logic I described is fairly straightforward to implement. For open you can just allow the user to pass an origin to the `open` method and it will default to `program` if they don't specify. That way if people really care about getting rid of the focus indicator they can add logic for it manually. It might also be nice to make a `mdSidenavTrigger` directive that does the focus origin logic for them.